includes/class-wc-stripe-feature-flags.php

```diff
@@ -189,4 +194,13 @@ public static function is_spe_available() {
 	public static function is_oc_available() {
 		return 'yes' === self::get_option_with_default( self::OC_FEATURE_FLAG_NAME );
 	}
+
+	/**
+	 * Checks whether the legacy checkout is deprecated.
+	 *
+	 * @return bool
+	 */
+	public static function is_legacy_checkout_deprecated() {
+		return WC_STRIPE_VERSION >= self::LEGACY_CHECKOUT_DEPRECATION_VERSION;
```

Comment:
```suggestion
		return version_compare( WC_STRIPE_VERSION, self::LEGACY_CHECKOUT_DEPRECATION_VERSION, '>=' );
```

Let's use `version_compare()` rather than a direct string numeric comparison. This should guards against weird edge cases, especially with patch versions.

Reply:
On second thoughts, why is the version comparison necessary? Could we not just return `true` since we will be depricating legacy in a major version that this change will be included in?

Reply:
This just allows us to ship this change before the 9.6 version. But I am OK with waiting for the specific version. In that case, instead of hardcoding the value, I would just not include this change at all (and [display the banner by default](https://github.com/woocommerce/woocommerce-gateway-stripe/pull/4328), for example).

Reply:
> This just allows us to ship this change before the 9.6 version

Ah thanks, that's the context I was missing. Let's just update it to use `version_compare` then.

Reply:
Alright. I applied the change already in https://github.com/woocommerce/woocommerce-gateway-stripe/pull/4382/commits/cc0d05f4d2758262c9436e625f2ee36c5e6bfdd2